output/cloud/expv2/sink.go

```diff
@@ -1,13 +1,78 @@
 package expv2
 
 import (
+	"fmt"
+
 	"go.k6.io/k6/metrics"
 )
 
-func newSink(mt metrics.MetricType) metrics.Sink {
-	if mt == metrics.Trend {
-		return &histogram{}
+type metricValue interface {
+	Add(v float64)
+}
+
+func newMetricValue(mt metrics.MetricType) metricValue {
+	var am metricValue
+	switch mt {
+	case metrics.Counter:
+		am = &counter{}
+	case metrics.Gauge:
+		am = &gauge{}
+	case metrics.Rate:
+		am = &rate{}
+	case metrics.Trend:
+		am = &histogram{}
+	default:
+		// Should not be possible to create
+		// an invalid metric type except for specific
+		// and controlled tests
+		panic(fmt.Sprintf("MetricType %q is not supported", mt))
+	}
+	return am
+}
+
+// TODO: add unit tests for the Add methods
+
+type counter struct {
+	Sum float64
+}
+
+func (c *counter) Add(v float64) {
+	c.Sum += v
+}
+
+type gauge struct {
+	Last     float64
+	Sum      float64
+	Min, Max float64
+	Avg      float64
+	Count    uint32
+
+	minSet bool
+}
```

Comment:
Moving here the discussion from https://github.com/grafana/k6/pull/3083#discussion_r1205527232

>Shouldn't this be just Last? Which is what Gauge basically is?

Reply:
Hey @esquonk, apparently other protocols (OTel, Prometheus remote write) don't support gauge in the same way. We may have issues if we want to migrate to them in the future. Considering this are not used at the moment, can we just drop them?

Reply:
We have those in the backend historically, and they are used. For example, if we have a "max" query for gauge metric, the code will look at "max" column.

We could just fill all the columns with "last" value at the cost losing a bit of data, but if we can collect some more statistical values, it make sense to do so. Unless it is too expensive to maintain.

As for compatibility with other protocols, I don't see how having more data internally can cause issues with that. If you need to produce an output in some protocol that only has `last`, you can just put last in there and ignore other values in your output.